src/wallet/mod.rs

```diff
@@ -454,6 +464,170 @@ impl Wallet {
+
+            builder
+                // important: the M5 OP_DRIVECHAIN output must come directly before the OP_RETURN sidechain address output.
+                .add_recipient(
+                    op_drivechain_output.script_pubkey,
+                    op_drivechain_output.value,
+                )
+                .add_data(&sidechain_address_data);
+
+            if let Some(fee_sats) = fee_sats {
+                builder.fee_absolute(fee_sats.to_sat());
+            }
+
+            if let Some((ctip_psbt_input, outpoint)) = ctip_foreign_utxo {
+                // This might be wrong. Seems to work!
+                let satisfaction_weight = 0;
+
+                builder
+                    .add_foreign_utxo(outpoint, ctip_psbt_input, satisfaction_weight)
+                    .map_err(WalletError)?;
+            }
+
+            builder.finish().into_diagnostic()?
+        };
+
+        Ok(psbt)
+    }
+
+    /// Creates a deposit transaction, persists it to the database, and returns the TXID.
+    /// This is also known as a M5 message, in BIP300 nomenclature.
+    ///
+    /// https://github.com/bitcoin/bips/blob/master/bip-0300.mediawiki#m5----deposit-btc-from-l1-to-l2
+    pub async fn create_deposit(
+        &self,
+        sidechain_number: SidechainNumber,
+        sidechain_address: String,
+        value_sats: Amount,
+        fee_sats: Option<Amount>,
+    ) -> Result<bdk::bitcoin::Txid> {
+        // If this is None, there's been no deposit to this sidechain yet. We're the first one!
+        let sidechain_ctip = self.validator.get_ctip(sidechain_number)?;
+        let sidechain_ctip = sidechain_ctip.as_ref();
+
+        let sidechain_ctip_amount = sidechain_ctip
+            .map(|ctip| ctip.value)
+            .unwrap_or(Amount::ZERO);
+
+        let op_drivechain_output = Self::create_deposit_op_drivechain_output(
+            sidechain_number,
+            sidechain_ctip_amount,
+            value_sats,
+        );
+
+        tracing::debug!(
+            "Created OP_DRIVECHAIN output: {}, {} ",
+            op_drivechain_output.value,
+            op_drivechain_output.script_pubkey.to_asm_string(),
+        );
+
+        let sidechain_address_bytes = sidechain_address.into_bytes();
+        let sidechain_address_data = PushBytesBuf::try_from(sidechain_address_bytes)
+            .map_err(|err| miette!("failed to convert sidechain address to PushBytesBuf: {err}"))?;
+
+        let psbt = self
+            .create_deposit_psbt(
+                op_drivechain_output,
+                sidechain_address_data,
+                sidechain_ctip,
+                fee_sats,
+            )
+            .await?;
+
+        tracing::debug!("Created deposit PSBT: {}", psbt.to_string());
+
+        let tx = self.sign_transaction(psbt)?;
+        let txid = tx.txid();
+
+        tracing::info!("Signed deposit transaction: `{}`", txid);
+
+        let mut tx_bytes = vec![];
+        tx.consensus_encode(&mut tx_bytes).into_diagnostic()?;
```

Comment:
You can use https://docs.rs/bitcoin/latest/bitcoin/consensus/fn.serialize.html here